File: Gateway/src/main/java/com/quizGpt/Gateway/Quiz/Entity/Question.java

package com.quizGpt.Gateway.Quiz.Entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@ToString
@Table(name = "Question")
public class Question {

    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long questionId;
    
    private String text;

    @OneToMany(targetEntity = Option.class, cascade = CascadeType.ALL)
    @JoinColumn( name="questionId_fk", referencedColumnName = "questionId")
    private List<Option> options;

    @OneToMany(targetEntity = Answer.class, cascade = CascadeType.ALL)
    @JoinColumn( name="questionId_fk", referencedColumnName = "questionId")
    private List<Answer> answers;
}

File: resource-server/src/quiz/entities/question.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  OneToMany,
  ManyToOne,
  Relation,
} from "typeorm";

import { Option } from "./option.js";
import { Answer } from "./answer.js";
import { Quiz } from "./quiz.js";

@Entity("Question")
export class Question {
  @PrimaryGeneratedColumn()
  questionId!: number;

  @Column()
  text!: String;

  @OneToMany(() => Option, (option) => option.question, { cascade: true })
  //   @JoinColumn({ name: "questionId_fk", referencedColumnName: "questionId" })
  options!: Relation<Option[]>;

  @OneToMany(() => Answer, (answer) => answer.question, { cascade: true })
  //   @JoinColumn({ name: "questionId_fk", referencedColumnName: "questionId" })
  answers!: Relation<Answer[]>;

  @ManyToOne(() => Quiz, (quiz) => quiz.questions)
  quiz!: Relation<Quiz>;
}
